### PR TITLE
feat: Support signed request to Elasticsearch service

### PR DESCRIPTION
**Pull request by h-kishi**
I added an implementation sending signed HTTP requests to Amazon Elasticsearch(OpenSearch) Service.
It is almost the same as [the official sample](https://docs.aws.amazon.com/opensearch-service/latest/developerguide/request-signing.html#request-signing-node).
It is compatible with the original behavior.

### CODE REVIEW

File: src/data-loaders/ElasticDataLoader.js [hunk 1]
```diff
@@ -7,20 +8,69 @@ export default class ElasticDataLoader {
...
+
+    const signer = new AWS.Signers.V4(httpRequest, 'es');
+    const credentials = {
+      accessKeyId: this.config.accessKeyId,
+      secretAccessKey: this.config.secretAccessKey,
+    };
+    signer.addAuthorization(credentials, new Date());
+
+    return httpRequest;
+  }
```
Comment (bboure) [inline src/data-loaders/ElasticDataLoader.js:75]: Could we also add the option to load the keys from envelope vats and/or local credentials file?

Comment (h-kishi) [reply src/data-loaders/ElasticDataLoader.js:72]: Good point.
I added `elasiticsearch.useSignature` option and modified to load credentials from config, env vars or local file.

By the way, I'm using the name `elasticsearch` in the config, would `openSearch` be more appropriate?

Comment (bboure) [reply src/data-loaders/ElasticDataLoader.js:72]: Great! thanks!

Yes `openSearch` is probably what we should start calling things now ✅

Comment (h-kishi) [reply src/data-loaders/ElasticDataLoader.js:72]: Thanks!
I changed the name from `elasticsearch` to `openSearch`.

**Comment by h-kishi** [comment]
@bboure
Do I need to take any other action?

**Comment by bboure** [comment]
@h-kishi sorry for the delay 

There is a merge conflict. 
Apart from that it looks good

**Comment by h-kishi** [comment]
@bboure Thank you for the reply.
I resolved the merge conflict.

**Comment by bboure** [comment]
LGTM

Thanks for the contrib!

**Pull request merged**